### PR TITLE
Improve use of defaults in validation tests

### PR DESCRIPTION
**Pull request by hayfield**
Brings various things in from `dev`, then uses them.

Primarily, see dd0138d8306628e4315820ec4f2e5706023302dd

### CODE REVIEW

File: iati/core/resources.py [hunk 1]
```diff
@@ -117,7 +116,7 @@ def get_all_schema_paths(version=None):
         Further exploration needs to be undertaken in how to handle multiple versions of the Standard.
```
Comment (allthatilk) [inline iati/core/resources.py]: Does this still need to be here (and other places in file)?

File: iati/core/tests/test_resources.py [hunk 2]
```diff
@@ -149,6 +139,13 @@ def test_load_as_dataset(self):
         assert isinstance(result, iati.core.Dataset)
         assert '<?xml version="1.0"?>\n\n<iati-activities version="2.02">' in result.xml_str
 
+    def test_load_as_dataset_invalid(self):
+        """Test that resources.load_as_dataset raises an error when the provided path does not lead to a file containing valid XML."""
```
Comment (allthatilk) [inline iati/core/tests/test_resources.py]: Does this need backticks?

File: iati/core/tests/test_resources.py [hunk 3]
```diff
@@ -158,6 +155,20 @@ def test_load_as_string(self):
         assert isinstance(result, six.string_types)
         assert result == 'This is a string that is not valid XML\n'
 
+    @pytest.mark.parametrize("load_method", [iati.core.resources.load_as_bytes, iati.core.resources.load_as_dataset, iati.core.resources.load_as_string])
+    def test_load_as_x_non_existing_file(self, load_method):
+        """Test that resources.load_as_bytes returns a bytes object with the expected content."""
```
Comment (allthatilk) [inline iati/core/tests/test_resources.py]: backticks? I'm still not clear on which things need them and which don't =/

File: iati/core/tests/test_schemas.py [hunk 4]
```diff
@@ -4,36 +4,43 @@
...
+        }
     ])
-    def schema_initialised(self, request):
+    def schema_initialised(self, request, standard_version_optional):
         """Create and return a single ActivitySchema or OrganisaionSchema object.
         For use where both ActivitySchema and OrganisaionSchema must produce the same result.
 
         Returns:
-            iati.core.ActivitySchema / iati.core.OrganisaionSchema: An activity and organisaion that has been initialised based on the default IATI Activity and Organisaion schemas.
+            iati.core.Schema: An activity or organisaion Schema that has been initialised.
```
Comment (allthatilk) [inline iati/core/tests/test_schemas.py]: organisaion -> organisation

File: iati/core/tests/test_schemas.py [hunk 5]
```diff
@@ -4,36 +4,43 @@
...
+            "schema_class": iati.core.ActivitySchema
+        },
+        {
+            "path_func": iati.core.resources.get_all_org_schema_paths,
+            "schema_class": iati.core.OrganisationSchema
+        }
     ])
-    def schema_initialised(self, request):
+    def schema_initialised(self, request, standard_version_optional):
         """Create and return a single ActivitySchema or OrganisaionSchema object.
```
Comment (allthatilk) [inline iati/core/tests/test_schemas.py]: OrganisaionSchema -> OrganisationSchema

File: iati/core/tests/test_schemas.py [hunk 6]
```diff
@@ -188,10 +204,18 @@ def test_schema_rulesets_add(self, schema_initialised):
 
     @pytest.mark.skip(reason='Not implemented')
     def test_schema_rulesets_add_twice(self, schema_initialised):
-        """Check that it is not possible to add the sameRulesets to a Schema multiple times."""
+        """Check that it is not possible to add the sameRulesets to a Schema multiple times.
```
Comment (allthatilk) [inline iati/core/tests/test_schemas.py]: sameRulesets -> same Rulesets

File: iati/core/tests/test_schemas.py [hunk 7]
```diff
@@ -188,10 +204,18 @@ def test_schema_rulesets_add(self, schema_initialised):
...
+
+        Todo:
+            Consider if this test should test against a versioned Ruleset.
```
Comment (allthatilk) [inline iati/core/tests/test_schemas.py]: Do we have other Ruleset versions?

File: iati/tests/test_validator.py [hunk 8]
```diff
@@ -337,14 +330,8 @@ def schema_version(self):
...
     def schema_org_type(self):
-        """A schema with the OrganisationType Codelist added.
-
-        Returns:
-            A valid activity schema with the OrganisationType Codelist added.
-
-        """
-        schema_path = iati.core.resources.get_schema_path(iati.core.tests.utilities.SCHEMA_NAME_VALID)
-        schema = iati.core.Schema(schema_path)
+        """Rturn an Activity Schema with the OrganisationType Codelist added."""
```
Comment (allthatilk) [inline iati/tests/test_validator.py]: Rturn -> Return

File: iati/core/tests/test_resources.py (hunk 2 above)
Comment (hayfield) [reply iati/core/tests/test_resources.py]: yep

File: iati/core/tests/test_resources.py (hunk 3 above)
Comment (hayfield) [reply iati/core/tests/test_resources.py]: References to other pieces of code should have them.

File: iati/core/tests/test_schemas.py (hunk 7 above)
Comment (hayfield) [reply iati/core/tests/test_schemas.py]: From a technical perspective, a Ruleset should be deemed as valid against a particular Decimal Version of the Standard. That they are all the same is, well, IATI for you.

**Comment by hayfield** [comment]
Changes made now

**Pull request merged**